nixos/tests/timezone.nix

```diff
@@ -1,45 +1,45 @@
 {
-  timezone-static = import ./make-test.nix ({ pkgs, ... }: {
+  timezone-static = import ./make-test-python.nix ({ pkgs, ... }: {
     name = "timezone-static";
     meta.maintainers = with pkgs.lib.maintainers; [ lheckemann ];
 
     machine.time.timeZone = "Europe/Amsterdam";
 
     testScript = ''
-      $machine->waitForUnit("dbus.socket");
-      $machine->fail("timedatectl set-timezone Asia/Tokyo");
-      my @dateResult = $machine->execute('date -d @0 "+%Y-%m-%d %H:%M:%S"');
-      $dateResult[1] eq "1970-01-01 01:00:00\n" or die "Timezone seems to be wrong";
+      machine.wait_for_unit("dbus.socket")
+      machine.fail("timedatectl set-timezone Asia/Tokyo")
+      date_result = machine.succeed('date -d @0 "+%Y-%m-%d %H:%M:%S"')
+      assert date_result == "1970-01-01 01:00:00\n", "Timezone seems to be wrong"
     '';
   });
 
-  timezone-imperative = import ./make-test.nix ({ pkgs, ... }: {
+  timezone-imperative = import ./make-test-python.nix ({ pkgs, ... }: {
     name = "timezone-imperative";
     meta.maintainers = with pkgs.lib.maintainers; [ lheckemann ];
 
     machine.time.timeZone = null;
 
     testScript = ''
-      $machine->waitForUnit("dbus.socket");
+      machine.wait_for_unit("dbus.socket")
 
       # Should default to UTC
-      my @dateResult = $machine->execute('date -d @0 "+%Y-%m-%d %H:%M:%S"');
-      print $dateResult[1];
-      $dateResult[1] eq "1970-01-01 00:00:00\n" or die "Timezone seems to be wrong";
+      date_result = machine.succeed('date -d @0 "+%Y-%m-%d %H:%M:%S"')
+      print(date_result)
+      assert date_result == "1970-01-01 00:00:00\n", "Timezone seems to be wrong"
 
-      $machine->succeed("timedatectl set-timezone Asia/Tokyo");
+      machine.succeed("timedatectl set-timezone Asia/Tokyo")
 
       # Adjustment should be taken into account
-      my @dateResult = $machine->execute('date -d @0 "+%Y-%m-%d %H:%M:%S"');
-      print $dateResult[1];
-      $dateResult[1] eq "1970-01-01 09:00:00\n" or die "Timezone was not adjusted";
+      date_result = machine.succeed('date -d @0 "+%Y-%m-%d %H:%M:%S"')
+      print(date_result)
+      assert date_result == "1970-01-01 09:00:00\n", "Timezone was not adjusted"
 
       # Adjustment should persist across a reboot
-      $machine->shutdown;
-      $machine->waitForUnit("dbus.socket");
-      my @dateResult = $machine->execute('date -d @0 "+%Y-%m-%d %H:%M:%S"');
-      print $dateResult[1];
-      $dateResult[1] eq "1970-01-01 09:00:00\n" or die "Timezone adjustment was not persisted";
+      machine.shutdown()
```

Comment:
can you make comments like "Adjustment should persist across a reboot" a subtest, so this information is even visible in the log?

Reply:
@tfc done